core/src/main/java/org/mule/runtime/core/exception/ErrorTypeRepository.java

```diff
@@ -7,105 +7,77 @@
+      ErrorTypeBuilder.builder().namespace(CORE_NAMESPACE_NAME).stringRepresentation(UNKNOWN_ERROR_IDENTIFIER)
+          .parentErrorType(ANY_ERROR_TYPE).build();
+
+  private Map<ComponentIdentifier, ErrorType> errorTypes = new HashMap<>();
 
   public ErrorTypeRepository() {
-    this.errorTypes.put(new ErrorTypeKey(CORE_NAMESPACE_NAME, ANY_IDENTIFIER),
+    this.errorTypes.put(new ComponentIdentifier.Builder()
+        .withNamespace(CORE_NAMESPACE_NAME)
+        .withName(ANY_IDENTIFIER)
+        .build(),
                         ANY_ERROR_TYPE);
+    this.errorTypes.put(new ComponentIdentifier.Builder()
+        .withNamespace(CORE_NAMESPACE_NAME)
+        .withName(UNKNOWN_ERROR_IDENTIFIER)
+        .build(),
+                        UNKNOWN_ERROR_TYPE);
 
   }
 
-  public void addErrorType(String namespace, String stringRepresentation, ErrorType parentErrorType) {
-    checkArgument(namespace != null, "namespace cannot be null");
-    checkArgument(stringRepresentation != null, "string representation cannot be null");
+  public void addErrorType(ComponentIdentifier errorTypeIdentifier, ErrorType parentErrorType) {
     ErrorTypeBuilder errorTypeBuilder =
-        ErrorTypeBuilder.builder().namespace(namespace).stringRepresentation(stringRepresentation)
+        ErrorTypeBuilder.builder().namespace(errorTypeIdentifier.getNamespace())
+            .stringRepresentation(errorTypeIdentifier.getName())
             .parentErrorType(parentErrorType);
-    if (this.errorTypes.put(new ErrorTypeKey(namespace, stringRepresentation), errorTypeBuilder.build()) != null) {
-      throw new IllegalStateException(format("An error type with namespace %s and string representation %s already exists",
-                                             namespace, stringRepresentation));
+    if (this.errorTypes.put(errorTypeIdentifier, errorTypeBuilder.build()) != null) {
+      throw new IllegalStateException(format("An error type with identifier %s", errorTypeIdentifier));
     }
   }
 
-  public ErrorType lookupErrorType(String namespace, String stringRepresentation) {
-    ErrorType errorType = this.errorTypes.get(new ErrorTypeKey(namespace, stringRepresentation));
+  public ErrorType lookupErrorType(ComponentIdentifier errorTypeComponentIdentifier) {
+    ErrorType errorType = this.errorTypes.get(errorTypeComponentIdentifier);
     if (errorType == null) {
-      throw new IllegalStateException(format("there's no error type with namespace %s and string representation", namespace,
-                                             stringRepresentation));
+      throw new IllegalStateException(format("there's no error type for %s", errorTypeComponentIdentifier));
     }
     return errorType;
   }
 
+  /**
+   * Gets the {@code ErrorType} instance for ANY error type.
+   *
+   * @return the ANY error type
+   */
   public ErrorType getAnyErrorType() {
     return ANY_ERROR_TYPE;
   }
 
-  public ErrorType getUnkonwnErrorType() {
```

Comment:
Was this deleted because it was not needed? Seems weird since unknown is defined here same as any.